src/stratis/errors.rs

```diff
@@ -66,7 +99,15 @@ impl fmt::Display for StratisError {
 impl Error for StratisError {
     fn cause(&self) -> Option<&dyn Error> {
         match *self {
-            StratisError::Error(_) | StratisError::Engine(_, _) => None,
+            StratisError::Msg(_) => None,
```

Comment:
While I agree updating to the `source` method makes sense here, my thought here with downcasting is that it may not actually be useful. You are expected to specify an explicit type to downcast to and this would result in `None` if the error is not actually of that type. The result of this would be a long set of if statements for all of our error matching attempting to downcast to each of our "accepted" errors and seeing which one returns `Some`. While maybe a trait object is what we want here, I am doubtful that downcasting will give us what we want. What are your thoughts?

Reply:
Seems like we're in agreement. ```source``` is the right way to go; benefits of downcasting uncertain, so let's not worry.